### ISSUE TITLE
`bacon test` doesn't recognize `--features`

### ISSUE THREAD

**Issue by nc7s** [Canop/bacon#171]
After running this:

```
lazy-regex
❯ bacon test --features std
```

I got this:

```
[lazy-regex] [test] [Command error code: 101]

    Finished test [unoptimized + debuginfo] target(s) in 0.01s
     Running unittests src/lib.rs (target/debug/deps/lazy_regex-53bac6e93dd32e9b)
error: Unrecognized option: 'features'
error: test failed, to rerun pass `--lib`
```

Putting `--features std` before `test`, i.e. `bacon --features std test`, doesn't work either.

This feels rather bizarre as the default job `check` runs fine with it.

**Comment by Canop** [comment]
What bacon does when you call it with `--features` is basically adding the argument at the end of the job command arguments.

Here's the default definition of the test job:

```
[jobs.test]
command = [
    "cargo", "test", "--color", "always",
    "--", "--color", "always", # see https://github.com/Canop/bacon/issues/124
]
```

The `--` prevents the `--features` from being given to the the cargo `test command`.

A workaround is to just edit your bacon.toml file to remove the second line of this command.

I should probably detect in bacon when there's a `--` and insert the `--features` arguments before.

**Comment by nc7s** [comment]
Thanks for the advice, it worked ;)

I see that the second line came from https://github.com/Canop/bacon/issues/124, cargo # 1983 being the main obstacle for a more elegant solution.

In a general sense, adding `--` to job commands, especially to default ones, feels like a bad choice: you don't know when that's needed, and "fixing" this one occasion means a full blown config file (speaking of that, maybe consider merging config with defaults rather than replacing?). If such arguments is absolutely necessary, maybe introduce a new field? Something like `passthru_args` (since they are "passed through to" artifacts).

**Comment by Canop** [comment]
Should be fixed now

**Comment by Canop** [comment]
(released in 2.15)
